cli/napi/threadsafe_functions.rs

```diff
@@ -226,65 +223,69 @@ fn napi_create_threadsafe_function(
   {
     return napi_generic_failure;
   }
-  *result = transmute::<Box<TsFn>, _>(Box::new(tsfn));
+  unsafe {
+    *result = transmute::<Box<TsFn>, _>(Box::new(tsfn));
+  }
 
   napi_ok
 }
 
-#[napi_sym::napi_sym]
+#[napi_sym]
 fn napi_acquire_threadsafe_function(
   tsfn: napi_threadsafe_function,
   _mode: napi_threadsafe_function_release_mode,
 ) -> napi_status {
-  let tsfn: &mut TsFn = &mut *(tsfn as *mut TsFn);
+  let tsfn: &mut TsFn = unsafe { &mut *(tsfn as *mut TsFn) };
   tsfn.acquire()
 }
 
-#[napi_sym::napi_sym]
+#[napi_sym]
 fn napi_unref_threadsafe_function(
   _env: &mut Env,
   tsfn: napi_threadsafe_function,
 ) -> napi_status {
-  let tsfn: &mut TsFn = &mut *(tsfn as *mut TsFn);
+  let tsfn: &mut TsFn = unsafe { &mut *(tsfn as *mut TsFn) };
   tsfn.unref()
 }
 
 /// Maybe called from any thread.
-#[napi_sym::napi_sym]
-pub fn napi_get_threadsafe_function_context(
+#[napi_sym]
+fn napi_get_threadsafe_function_context(
   func: napi_threadsafe_function,
   result: *mut *const c_void,
 ) -> napi_status {
-  let tsfn: &TsFn = &*(func as *const TsFn);
-  *result = tsfn.context;
+  let tsfn: &TsFn = unsafe { &*(func as *const TsFn) };
+  unsafe {
+    *result = tsfn.context;
+  }
   napi_ok
 }
 
-#[napi_sym::napi_sym]
+#[napi_sym]
 fn napi_call_threadsafe_function(
   func: napi_threadsafe_function,
   data: *mut c_void,
   is_blocking: napi_threadsafe_function_call_mode,
 ) -> napi_status {
-  let tsfn: &TsFn = &*(func as *const TsFn);
-  tsfn.call(data, is_blocking != 0);
+  let tsfn: &TsFn = unsafe { &*(func as *const TsFn) };
+  tsfn.call(data, is_blocking == napi_tsfn_blocking);
   napi_ok
 }
 
-#[napi_sym::napi_sym]
+#[napi_sym]
 fn napi_ref_threadsafe_function(
   _env: &mut Env,
   func: napi_threadsafe_function,
 ) -> napi_status {
-  let tsfn: &mut TsFn = &mut *(func as *mut TsFn);
+  let tsfn: &mut TsFn = unsafe { &mut *(func as *mut TsFn) };
   tsfn.ref_()
 }
 
-#[napi_sym::napi_sym]
+#[napi_sym]
 fn napi_release_threadsafe_function(
   tsfn: napi_threadsafe_function,
   _mode: napi_threadsafe_function_release_mode,
 ) -> napi_status {
-  let tsfn: Box<TsFn> = Box::from_raw(tsfn as *mut TsFn);
+  let tsfn: Box<TsFn> = unsafe { Box::from_raw(tsfn as *mut TsFn) };
```

Comment:
I like that `napi_sym` is no longer accepting all unsafe code and requires explicit blocks 👍